qa/L0_backend_python/response_sender/response_sender_test.py

```diff
@@ -0,0 +1,293 @@
+            send_complete_final_flag_before_return=True,
+            return_a_response=False,
+            number_of_response_after_return=0,
+            send_complete_final_flag_after_return=False,
+        )
+
+    # Decoupled model send response final flag after request return.
+    def test_decoupled_zero_response_post_return(self):
+        self._assert_decoupled_infer_success(
+            number_of_response_before_return=0,
+            send_complete_final_flag_before_return=False,
+            return_a_response=False,
+            number_of_response_after_return=0,
+            send_complete_final_flag_after_return=True,
+        )
+
+    # Decoupled model send 1 response before request return.
+    def test_decoupled_one_response_pre_return(self):
+        self._assert_decoupled_infer_success(
+            number_of_response_before_return=1,
+            send_complete_final_flag_before_return=True,
+            return_a_response=False,
+            number_of_response_after_return=0,
+            send_complete_final_flag_after_return=False,
+        )
+
+    # Decoupled model send 1 response after request return.
+    def test_decoupled_one_response_post_return(self):
+        self._assert_decoupled_infer_success(
+            number_of_response_before_return=0,
+            send_complete_final_flag_before_return=False,
+            return_a_response=False,
+            number_of_response_after_return=1,
+            send_complete_final_flag_after_return=True,
+        )
+
+    # Decoupled model send 2 response before request return.
+    def test_decoupled_two_response_pre_return(self):
+        self._assert_decoupled_infer_success(
+            number_of_response_before_return=2,
+            send_complete_final_flag_before_return=True,
+            return_a_response=False,
+            number_of_response_after_return=0,
+            send_complete_final_flag_after_return=False,
+        )
+
+    # Decoupled model send 2 response after request return.
+    def test_decoupled_two_response_post_return(self):
+        self._assert_decoupled_infer_success(
+            number_of_response_before_return=0,
+            send_complete_final_flag_before_return=False,
+            return_a_response=False,
+            number_of_response_after_return=2,
+            send_complete_final_flag_after_return=True,
+        )
+
+    # Decoupled model send 1 and 3 responses before and after return.
+    def test_decoupled_response_pre_and_post_return(self):
+        self._assert_decoupled_infer_success(
+            number_of_response_before_return=1,
+            send_complete_final_flag_before_return=False,
+            return_a_response=False,
+            number_of_response_after_return=3,
+            send_complete_final_flag_after_return=True,
+        )
+
+    # Non-decoupled model send 1 response on return.
+    def test_non_decoupled_one_response_on_return(self):
+        self._assert_non_decoupled_infer_success(
```

Comment:
All these tests are looking for successes. Do we have any test cases to assert failures happen?

Reply:
Not added yet, they will be added later in the series of PRs.